### PR TITLE
Feature/cloud backups

### PR DESCRIPTION
**Pull request by dr-bonez**

### CODE REVIEW

**Comment by dr-bonez** [comment]
Ok, this is ready for review, only review the last commit, the rest is because it was based on bugfix/disable-uasp.
Once #878 is merged, I will do an interactive rebase to clean it.

**Comment by dr-bonez** [comment]
ux quirks:
- Hostname should be just the name of the computer. No `smb://` no `.local`. The placeholder text should be changed to reflect this. Ideally, also strip that stuff out if they put it.
- `\` should be replaced with `/` in the `path` field

## Creating a shared folder

### Windows
1. Create a folder
2. Right click the folder
3. select "properties"
4. click the `share` tab
5. click `share...`
6. select a user you want to use for login, or select `everyone`
7. click `share`
8. enter your computer name into the `hostname` field in embassy ui (the name is shown after a `\\` in the windows ui)
9. enter the path that is shown after the `hostname` in the windows ui into the `path` field
10. enter your username for the windows computer
11. enter the password for the windows computer

### Mac
1. go to system settings
2. click `sharing`
3. click `file sharing`
4. click the `+` under `shared folders`
5. add a folder
6. click options
7. enable `Windows file sharing` for the user you would like to use to authenticate
8. click done
9. enter the name of your macbook, as shown in the `computer name` field at the top of the `sharing` page into the `hostname` field in embassy ui
10. enter the name of the folder as shown in the `shared folders` section as the `path`
11. enter your macbook username and password for the user that you enabled `windows file sharing` for

**Comment by chrisguida** [comment]
Tested ACK

File: appmgr/src/db/model.rs [hunk 1]
```diff
@@ -32,7 +32,12 @@ pub struct Database {
...
 }
 impl Database {
-    pub fn init(id: String, hostname: &str, tor_key: &TorSecretKeyV3) -> Self {
+    pub fn init(
+        id: String,
+        hostname: &str,
+        tor_key: &TorSecretKeyV3,
+        password_hash: String,
+    ) -> Self {
         // TODO
```
Comment (ProofOfKeags) [inline appmgr/src/db/model.rs:41]:
```suggestion
```

**Review by ProofOfKeags — APPROVED** [review]

**Review by ProofOfKeags — APPROVED** [review]

**Review by MattDHill — APPROVED** [review]

**Pull request merged**